### PR TITLE
Adding feature to fix mirror if any missing crates are found.

### LINKED ISSUE #64: Missing crates not downloaded when syncing mirror
Hi there :wave: !
First of all, well done for your work on `panamax` !

So, the issue here has already been highlighted in others issues (same author):

- [issue#55](https://github.com/panamax-rs/panamax/issues/55): the issue's author seemed to have an incoherent mirror. To solve it, he had to delete his actual mirror and sync a new mirror from scratch. I already met this issue, and had to go through the same process. But is is a costly solution (in my opinion), and the cost will only grow as `crates.io` will reference more and more crates.
- [issue#8](https://github.com/panamax-rs/panamax/issues/8)): author used the same fix here.

I opened a pull request in January to add a feature to `panamax` allowing users to check for missing crates in the synced mirror [PR#57](https://github.com/panamax-rs/panamax/pull/57).
The check is not fetching the `crates.io-index`, but just looking at the the local index state.

I closed it as I had no feedback from you. But I decided to still open an issue about it.

I couldn't find a way to trigger the problem using `panamax`, but I decided to manually delete some crates from my local mirror.
Here are the steps I did (from my [`panamax`'s fork](https://github.com/ricked-twice/panamax)):

1. Removed `/path/to/my/mirror/crates/fa/st/fastrand/` and `/path/to/my/mirror/crates/ri/ng/ring/0.16.20/`
2. Ran `cargo run -- verify /path/to/my/mirror`. The manually deleted crate are spotted by the check.
3. Ran `cargo run -- sync /path/to/my/mirror`. Everything went find according to the execution's output.
4. Ran `cargo run -- verify /path/to/my/mirror`. The manually deleted crate are still spotted by the check.
5. Ran `ls /path/to/my//mirror/crates/fa/st/fastrand /path/to/my//mirror/crates/ri/ng/ring/0.16.20` just to be sure. And get `No such file or directory` error for both directory.

There are two things highlighted by this issue:

1. If, for some reason, some crates are deleted for the mirror, `panamax` is not able to download them again because of the fact that the reference to download crates is a diff between the local index (`master` branch) and the distant one (`origin/master` branch). This makes sense that the missing crates are not downloaded.
2. That behavior does not explain why the mirror can sometimes be in an inconsistent state without any crates removal.

By exchanging about it around me, I've been put on the track that it could come from the fact that our local commit have been squashed on origin which result of local commit id to be absent from the tree. But I've looked into `panamax`'s code and you seem to handle this case. So, I can't figure a reason of this behavior.

Anyway, I hope this issue will lead to some discussion, and if you need help, I'd be glad to re-open a PR.

Cheers :smile:

### PR DESCRIPTION
**Pull request by ricked-twice**
This PR should definitely fix #64.

This is a huge single commit.

TLDR:
- Modified `Panamax::Verify` variant to take two options:
   - `--dry_run`: whether to fix or just check mirror state;
   - `--assume_yes`: allowing users to skip verification phase before downloading anything.
- Modified `mirror::verify` to take those arguments into account;
- Modification on `verify`:
   - Added `Input` enum to handle user's input through `FromStr` trait;
   - Added tests to check `FromStr` implementation;
   - Removed `VerifyError` enum;
   - Modified `verify_mirror` to return `Result<Option<Vec>, MirrorError>`;
   - Added `handle_user_input` function to let user filter or not missing crates before downloading them;
   - Added `fix_mirror` function to actually download missing crates.

----------------------------------

The `verify` feature is now able to download any missing crates.
By default it is asking user's confirmation before downloading any found missing crates and allow them to filter missing crates to download.

It can take several options:
```bash
# only checking if missing crates are found and print them to stdout
$ panamax verify --dry-run /path/to/mirror
# fixing mirror if needed without any user's interaction
$ panamax verify --assume-yes /path/to/mirror
```

### CODE REVIEW

**Comment by JamesMConroy** [comment]
I'm using this branch and haven't run into any issues yet.

**Comment by k3d3** [comment]
LGTM, thank you!
